casbin/model/Model.h

```diff
@@ -28,7 +28,7 @@ using namespace std;
 class AssertionMap {
     public:
 
-        unordered_map<string, Assertion*> assertion_map;
+        unordered_map<string, shared_ptr<Assertion>> assertion_map;
```

Comment:
Do we really need a smart pointer here? how about using an owned `Asertion` here since it wan't be used elsewhere.

Reply:
> Do we really need a smart pointer here? how about using an owned `Asertion` here since it wan't be used elsewhere.

I think we have two choices:
 1.use unordered_map<string, shared_ptr<Assertion>> assertion_map; as Divy,and new assertion to insert.
2.unordered_map<string, Assertion>,assign the value of assertion to insert.
I can't judge which one is better.